src/bindings/python/valhalla/actor.py

```diff
@@ -0,0 +1,115 @@
+import json
+import tempfile
+from pathlib import Path
+from typing import Union
+
+from ._valhalla import _Actor
+
+
+# TODO: wasteful for dict input/output; more reasonable would be to extend
+#   the Actor's action C++ interfaces with a JSON arg
+def dict_or_str(func):
+    def wrapped(*args):
+        # /status doesn't take any parameters
+        if not len(args) > 1:
+            return func(*args)
+
+        if isinstance(args[1], dict):
+            return json.loads(func(args[0], json.dumps(args[1])))
+        elif not isinstance(args[1], str):
+            raise ValueError("Request must be either of type str or dict")
+        return func(*args)
+
+    return wrapped
+
+
+class Actor(_Actor):
+    def __init__(self, config: Union[Path, str, dict]):
```

Comment:
I also added this whole `__init__`. basically it allows to config the actor with a dict and does some more validation, making sure there's at least an existing tile_dir